src/Nethermind/Nethermind.Serialization.Json/UInt256Converter.cs

```diff
@@ -29,8 +30,19 @@ private static UInt256 ReadInternal(ref Utf8JsonReader reader, JsonTokenType all
     {
         if (reader.TokenType == JsonTokenType.Number)
         {
-            return reader.GetUInt64();
+            ReadOnlySpan<byte> bytes = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+
+            if (Utf8Parser.TryParse(bytes, out ulong shortValue, out int bytesConsumed) && bytes.Length == bytesConsumed)
+            {
+                return new UInt256(shortValue);
+            }
+
+            Span<char> chars = stackalloc char[bytes.Length + 1];
```

Comment:
Unbounded stack allocation from user input; means some could take down the server by adding very long numeric values to UInt256 fields

Reply:
Also if variable length, move to different method as will make the parsing of hex UInt256 horrible even when this isn't reached

Reply:
Yep, let's make it safer. Not sure about separate method, like something like in the latest commit?